applications/StructuralMechanicsApplication/python_scripts/structural_response.py

```diff
@@ -67,26 +68,32 @@ def __init__(self, identifier, response_settings, model_part = None):
         with open(response_settings["primal_settings"].GetString()) as parameters_file:
             ProjectParametersPrimal = Parameters(parameters_file.read())
 
-        self.primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(ProjectParametersPrimal, model_part)
-        self.primal_analysis.GetModelPart().AddNodalSolutionStepVariable(SHAPE_SENSITIVITY)
+
+        self.primal_model_part = model_part
+        model = Model()
+        model.AddModelPart(self.primal_model_part)
+        self.primal_analysis = structural_mechanics_analysis.StructuralMechanicsAnalysis(model, ProjectParametersPrimal)
+        self.primal_model_part.AddNodalSolutionStepVariable(SHAPE_SENSITIVITY)
 
     def Initialize(self):
         self.primal_analysis.Initialize()
         self.response_function_utility.Initialize()
 
     def InitializeSolutionStep(self):
-        self.primal_analysis.InitializeTimeStep()
+        self.primal_analysis.time = self.primal_analysis.solver.AdvanceInTime(self.primal_analysis.time)
```

Comment:
This seems hacky...

Reply:
We have to speak abt how to integrate the StructuralAnalysis properly, riccardo doesn't want us to use it in this way, I will explain to you in person